perf-test/components/SkottiePlayer/index.tsx

```diff
@@ -17,8 +17,11 @@ interface Props {
 }
 
 export default function Skottie ({ lottieURL, width, height }: Props) {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
   const [id, setId] = useState<string>('');
+  let [ratio, setRatio] = useState<number>(1.0);
   let initialized = false;
+  let changingRatio = false;
```

Comment:
this looks unnecessary?

Reply:
Without this, Yes it works. But to change DPR with canvasKit, calling `MakeSWCanvasSurface` is required after DPR changed, this call takes seconds to process them.

So I define this status to avoid duplicated calling while canvas is built. Because `drawFrame` is called every frame, I use this condition for the duplicated processing.

```js
if (dpr !== ratio && !changingRatio) {
```

Reply:
But your logic looks wiered. changingRatio is only toggled in the scope? Is it correct?

Reply:
`drawFrame()` is parallelly called. When `MakeSWCanvasSurface` hangs on, another `drawFrame()` newly called.

If changingRatio is `true`, we can skip these process in newly called callback.

Reply:
it sounds like the context can be broken. should be never happend.

Reply:
Actually we don't need this logic. We just use system default value for DPR. Don't need to think of change in time.